src/preprocessing.py

```diff
@@ -0,0 +1,443 @@
+        test_range (Tuple[int, int]): The range of GitHub IDs for the test set.
+
+    Returns:
+        Tuple[pd.DataFrame, pd.DataFrame]:
+            - pd.DataFrame: The training set DataFrame.
+            - pd.DataFrame: The test set DataFrame.
+
+    Raises:
+        ValueError: If 'github_id' column is missing in the DataFrame.
+    """
+    if 'github_id' not in issues_df.columns:
+        raise ValueError("The DataFrame must contain a 'github_id' column.")
+
+    train_start, train_end = train_range
+    test_start, test_end = test_range
+
+    train_set = issues_df[(issues_df['github_id'] >= train_start) & (issues_df['github_id'] <= train_end)]
+    test_set = issues_df[(issues_df['github_id'] >= test_start) & (issues_df['github_id'] <= test_end)]
+    return train_set, test_set
+
+def save_data(dataset_df: pd.DataFrame, path: str) -> None:
+    """
+    Save the DataFrame as a JSON file at the specified path.
+
+    Args:
+        dataset_df (pd.DataFrame): The DataFrame to be saved.
+        path (str): The file path where the JSON file will be saved.
+
+    Returns:
+        None
+
+    Raises:
+        IOError: If the file cannot be written.
+    """
+    try:
+        os.makedirs(os.path.dirname(path), exist_ok=True)
+        dataset_df.to_json(path, orient='records', indent=2)
+    except IOError as e:
+        raise IOError(f"Failed to save data to {path}: {e}")
+
+def main() -> None:
+    """
+    Main function to execute the data preprocessing pipeline.
+
+    Steps:
+        1. Pull issues from the specified GitHub repository.
+        2. Remove issues with infrequent assignees.
+        3. Preprocess issues by extracting markdown elements and cleaning text.
+        4. Split data into training and test sets.
+        5. Save the datasets to JSON files.
+
+    Raises:
+        Exception: If any step in the pipeline fails.
+    """
+    try:
+        from pull_issues import pull_issues
+        
+        issues_df = pull_issues("microsoft/vscode")
+        assert isinstance(issues_df, pd.DataFrame), f"Expected 'issues_df' to be a DataFrame but got {type(issues_df)}"
+
+        issues_df = remove_infrequent_assignees(issues_df, min_assignments=50)
+
+        issues_df = preprocess_issues(issues_df)
+
+        train_range = (1, 210000)
+        test_range = (210001, 220000)
+        train_set, test_set = split_data(issues_df, train_range, test_range)
+
+        train_path = os.path.join('data', 'train', 'train_issues.json')
+        test_path = os.path.join('data', 'test', 'test_issues.json')
+
+        print(f"\nSaving training dataset to {train_path} with {train_set.shape[0]} issues")
+        save_data(train_set, train_path)
+
+        print(f"Saving test dataset to {test_path} with {test_set.shape[0]} issues")
+        save_data(test_set, test_path)
+    except Exception as e:
+        print(f"An error occurred during preprocessing: {e}")
+        raise
```

Comment:
?????

Why have a `try except` only to print something and to raise the exception again??
It literally does nothing more than the standard behaviour of python.....
Even worse, you print the error to `stdout` instead of the normal `stderr` and you raise nothing......


Like, do you even read what ChatGPT gives you???